packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts

```diff
@@ -513,6 +513,487 @@ export class CloudFormationDeleteStackAction extends CloudFormationDeployAction
+   *
+   * @default SELF_MANAGED
+   */
+  readonly permissionModel?: StackSetPermissionModel;
+
+  /**
+   * Note
+   * Because AWS CloudFormation StackSets performs operations across multiple accounts, you must define the necessary permissions in those accounts
+   * before you can create the stack set.
+   *
+   * Note
+   * This parameter is optional for the SELF_MANAGED permissions model and is not used for the SERVICE_MANAGED permissions model.
+   *
+   * The IAM role in the administrator account used to perform stack set operations.
+   *
+   * If you do not specify the role, it is set to AWSCloudFormationStackSetAdministrationRole. If
+   * you specify ServiceManaged, you must not define a role name.
+   *
+   * @default AWSCloudFormationStackSetAdministrationRole
+   */
+  readonly administrationRole?: iam.IRole;
+
+  /**
+   * Note
+   * Because AWS CloudFormation StackSets performs operations across multiple accounts, you must define the necessary permissions in those accounts
+   * before you can create the stack set.
+   *
+   * Note
+   * This parameter is optional for the SELF_MANAGED permissions model and is not used for the SERVICE_MANAGED permissions model.
+   *
+   * The name of the IAM role in the target accounts used to perform stack set operations. The name may contain alphanumeric characters, any of the
+   * following characters: _+=,.@-, and no spaces. The name is not case sensitive. This role name must be a minimum length of 1 character and maximum
+   * length of 64 characters. Role names must be unique within the account. The role name specified here must be an existing role name. Do not specify
+   * this role if you are using customized execution roles. If you do not specify the role name, it is set to AWSCloudFormationStackSetExecutionRole.
+   * If you set Service_Managed to true, you must not define a role name.
+   *
+   * @default AWSCloudFormationStackSetExecutionRole
+   */
+  readonly executionRoleName?: string;
+
+  /**
+   * Note
+   * This parameter is optional for the SERVICE_MANAGED permissions model and is not used for the SELF_MANAGED permissions model.
+   *
+   * Describes whether AWS CloudFormation StackSets automatically deploys to AWS Organizations accounts that are added to a target organization or
+   * organizational unit (OU). If OrganizationsAutoDeployment is specified, do not specify DeploymentTargets and Regions.
+   *
+   * Note
+   * If no input is provided for OrganizationsAutoDeployment, then the default value is Disabled.
+   *
+   * @default Disabled
+   */
+  readonly organizationsAutoDeployment?: StackSetOrganizationsAutoDeployment;
+
+  /**
+   * Note
+   * For the SERVICE_MANAGED permissions model, you can provide either accounts or organizational Unit IDs for deployment targets. For the
+   * SELF_MANAGED permissions model, you can only provide accounts.
+   *
+   * Note
+   * When this parameter is selected, you must also select Regions.
```

Comment:
Again, this constraint means we are missing a higher-level concept here that probably deserves its own class, and the `deploymentTargets` and `regions` properties should be merged.